pkg/services/baremetal/host/host.go

```diff
@@ -588,6 +588,31 @@ func (s *Service) actionRegistering() actionResult {
 		return s.recordActionFailure(infrav1.RegistrationError, err.Error())
 	}
 
+	// Check RAID for the second time.
+	// This happens for new machines which don't have rootDeviceHints.
+	// See "tworaidchecks" for the other place.
+	msg = ""
+	if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage == nil {
+		msg = "Internal Error: HetznerBareMetalHost.Spec.Status.InstallImage is nil. setHostSpec() not called yet?"
+	} else if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Swraid != 0 &&
+		len(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.Raid.WWN) < 2 {
+		msg = "HetznerBareMetalHost.Spec.Status.InstallImage.Swraid is active. List spec.rootDevideHints.raid.wwn must contain at least two entries."
+	} else if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Swraid == 0 &&
+		len(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.Raid.WWN) > 0 {
```

Comment:
what we actually want to check here is that rootDeviceHints.wwn is not empty, don't we?

Reply:
Some lines above `IsValidWithMessage()` gets called. There we check that single-WWN  XOR  raid-WWNs are specified. So it does not matter if I check the Raid.WWN list or the single WWN string.

But I can change that, if you prefer the other.

Reply:
yes I think that makes it easier to read. I am aware that it comes down to the same thing

Reply:
@guettli I think you now did it the other way around and also made the other case indirect. I would prefer the direct one for both! 

If you are supposed to specifiy a raid, it should be "specify a raid" and not "specify not a wwn". And the other way around

Reply:
@janiskemper the two error messages are now:

> Invalid HetznerBareMetalHost: spec.status.installImage.swraid is active and wwn="wwn1". Use spec.rootDevideHints.raid.wwn instead of spec.rootDevideHints.wwn.

> Invalid HetznerBareMetalHost: spec.rootDevideHints.raid.wwn ([wwn1 wwn2]) contains several WWNs, but spec.status.installImage.swraid is not active.

If you want it different, then please tell me exactly your desired err msg.

Reply:
@janiskemper I implemented option-1, please speak up if you want something else.

Reply:
I described it above, that I think you should do the "direct" check and not the "indirect" one. I understand that it comes down to the same thing because we have the XOR operator, but I do think that it makes the code and the error message better.

If you don't know what I mean, then let's talk about it

Reply:
@janiskemper it feels exhausting for me to guess what you want. Please take the PR and adapt the error message to the way you want it to be. Or post a code snippet here, and I will update the PR.